add the list authorization of policy-encryption-key for governance-po…

yanmxa opened the pull request:
[…licy-spec-sync](https://github.com/open-cluster-management-io/governance-policy-spec-sync/issues/26)

JustinKuli commented:
Thanks for reporting this issue! This will need to be added to the permission in the kubebuilder tag here: https://github.com/open-cluster-management-io/governance-policy-spec-sync/blob/main/controllers/secretsync/secret_sync.go#L46, instead of modifying the operator.yaml directly.

I'm surprised that the existing e2e test isn't failing because of this. Maybe we need a test just to ensure the container is stable when deployed, but I wouldn't require that in this PR.

clyang82 commented:
ref: https://github.com/open-cluster-management-io/clusteradm/issues/210

yanmxa commented:
> Thanks for reporting this issue! This will need to be added to the permission in the kubebuilder tag here: https://github.com/open-cluster-management-io/governance-policy-spec-sync/blob/main/controllers/secretsync/secret_sync.go#L46, instead of modifying the operator.yaml directly.
> 
> I'm surprised that the existing e2e test isn't failing because of this. Maybe we need a test just to ensure the container is stable when deployed, but I wouldn't require that in this PR.

I have added the permission in the kubebuilder tag. Thanks for the reminder.

Although the governance-policy-spec-sync pod keeps restarting. It will also go into Running status for a little while, and this short period of Running status may allow it to escape the e2e  test.

yanmxa commented:
> @yanmxa thanks for catching this! Could you please ensure your commit message matches the contribution guidelines? https://github.com/open-cluster-management-io/community/blob/main/sig-policy/contribution-guidelines.md#commit-messages

Thanks for your information. I added the signoff in the later comment.

JustinKuli commented:
The test failure is (probably) just because the permissions file was modified directly, and when it's generated it's a little different:
```
diff --git a/deploy/operator.yaml b/deploy/operator.yaml
index 90864cc..258bf2e 100644
--- a/deploy/operator.yaml
+++ b/deploy/operator.yaml
@@ -82,8 +82,8 @@ rules:
   verbs:
   - delete
   - get
-  - update
   - list
+  - update
 - apiGroups:
   - policy.open-cluster-management.io
   resources:
diff --git a/deploy/rbac/role.yaml b/deploy/rbac/role.yaml
index 030257c..aba7409 100644
--- a/deploy/rbac/role.yaml
+++ b/deploy/rbac/role.yaml
@@ -40,6 +40,7 @@ rules:
   verbs:
   - delete
   - get
+  - list
   - update
 - apiGroups:
   - policy.open-cluster-management.io
```

mprahl commented:
Could you please run `make generate-operator-yaml` and then commit the changed YAML files in `deploy`? Then could you also please squash the commits?

yanmxa commented:
> Could you please run `make generate-operator-yaml` and then commit the changed YAML files in `deploy`? Then could you also please squash the commits?

Hey @mprahl the controller-gen seems cann't be install to the sepecified directory when I run `make generate-operator-yaml` in the project.
```
$ go version    
go version go1.18.1 linux/amd64
$ pwd
/root/workspace/ocm/governance-policy-spec-sync
$ make generate-operator-yaml                                                   
go: creating new go.mod: module tmp
Downloading sigs.k8s.io/kustomize/kustomize/v3@v3.8.7
go: downloading sigs.k8s.io/kustomize/kustomize/v3 v3.8.7
go: downloading k8s.io/client-go v0.18.10

...

go: added sigs.k8s.io/structured-merge-diff/v4 v4.1.0
go: added sigs.k8s.io/yaml v1.2.0
/root/workspace/ocm/governance-policy-spec-sync/bin/controller-gen  rbac:roleName=governance-policy-spec-sync paths="./..." output:rbac:artifacts:config=deploy/rbac
/bin/sh: /root/workspace/ocm/governance-policy-spec-sync/bin/controller-gen: No such file or directory
make: *** [Makefile:163: manifests] Error 127
```
